provider.go

```diff
@@ -23,37 +60,23 @@ func configureProvider(log logr.Logger, provider *plugin.GRPCProvider) {
 	//    When these items are missing the error messages are of the format `features attribute is required`
 	// 3. When the `cty.ObjectVal` doesn't follow the required schema the error messages provided back don't make this immediately clear.
 	//    You may for example receive a message of `attribute 'use_msi' bool is required` when the error was introducing the wrong structure for the `features` list
-	configProvider := providerConfigBlock.EmptyValue()
-
-	// Here is an example of a list min 1.
-	// The `features` block in the Azure RM provider
-	//
-	// provider "azurerm" {
-	// 	version = "=2.0.0"
-	// 	features {}
-	// }
-	//
-	// Represented as YAML this would be:
-	//
-	// features:
-	// - ~
-
-	// Workaround to create a `cty.ListVal` for `features` block with one blank item in it.
-	// Get block definition
-	featuresType := providerConfigBlock.BlockTypes["features"]
-	// Create a map to represent the block
-	featuresBlockMap := map[string]cty.Value{}
-	log.Info("%v", featuresType)
-	// Get each of the nested blocks required in the block and create
-	// empty items for them. Insert them into the featuresBlockMap
-	for name, nestedBlock := range featuresType.BlockTypes {
-		featuresBlockMap[name] = nestedBlock.EmptyValue()
+	providerConfigBlock := provider.GetSchema().Provider.Block
+
+	// Parse the content of the provider block given to us into a body.
+	file, diagParse := hclsyntax.ParseConfig([]byte(configBody), "test.tf", hcl.Pos{})
```

Comment:
Is the filename (`test.tf`) important here? If not, how about `dummy.tf`?

Reply:
Great spot, yeah exactly right - will update to `dummy.tf`